src/components/Footer.jsx

```diff
@@ -80,16 +79,18 @@ class Footer extends React.Component {
     sendTrackEvent(eventName, properties);
   }
 
-  CCPADialogCloseHandler() {
-    this.setState({
-      showCCPADialog: false,
-    });
-  }
-
-  CCPADialogOpen() {
-    this.setState({
-      showCCPADialog: true,
-    });
+  toggleConsentModal() {
+    window.OneTrust?.ToggleInfoDisplay();
+    const { host } = new URL(`${MARKETING_BASE_URL}`);
```

Comment:
The logic per se looks good at first glance. However, this looks like you are trying to extract relevant parts from the MARKETING_BASE_URL constant. Since that is a constant and cannot change, this may be adding unnecessary complexity. Are you trying to grab the string "edx.org" for a cookie? I agree that having this here as a magical string in the method is not the best way, but perhaps simply adding a second constant above `const MARKETING_BASE_URL = ...` on line 28 is a nice alternative? In that case, a lot of this code could be removed, and the lower complexity would decrease the risk of any bugs in the code later. Something like `const domain = MARKETING_DOMAIN` will be fine.

Correct me if I'm wrong here. The constructor for `URL` just takes 2 arguments, a url and an optional base. Using both arguments would construct a full url which then would make sense to split and grab parts from, but since you are only providing the one argument, it seems - from my test in my browser - just always to return host: 'edx.org'.